File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        a = {}
        for i in range(len(s)):
            if s[i] not in a:
                a[s[i]] = [i]
            else:
                a[s[i]].append(i)
        for i in a:
            if len(a[i])==1:
                return a[i][0]
        return -1

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char,int> map;
        for(int i=0; i<s.length(); i++){
            if(map.count(s[i])==0)
                map[s[i]] = 1;
            else
                map[s[i]]++;   
        }
        for(int i=0; i<s.length(); i++)
            if(map[s.at(i)] == 1)
                return i;
        return -1;
        //-------------Another approach---------------------------------------
		// vector<int> v(26,0);
		// for(char c : s) v[c - 'a']++;
		// for(int i = 0; i < s.length(); i++){
		// 	if(v[s[i] - 'a'] == 1) return i;
		// }
		// return -1;
    }
};